docs/design-docs/cli.md

```diff
@@ -0,0 +1,51 @@
+# Studio CLI
+
+## About this doc
+
+This document outlines the design and implementation details for the Studio CLI utility. It covers the high-level approach, data flows, and implementation details for this feature.
+
+## Context
+
+The Studio CLI (invoked with the `studio` command) is a globally available CLI utility allowing users to interact with various Studio features independently of the desktop application.
+
+## High level approach
+
+The CLI is independent of the main desktop app, but is written using mostly the same conventions. It's a node.js app written in Typescript, that's bundled into a single JS file using Webpack. Jest is used to test CLI modules in the same way as for regular Studio modules.
+
+To run the CLI, we first add a script to a directory on `$PATH`. This script runs the CLI JS file using the node runtime bundled with Studio. Running JS files independently of the main Studio app is possible thanks to the `ELECTRON_RUN_AS_NODE=1` option.
+
+The first iteration of the CLI shipped commands to create, read, update, and delete preview sites. To keep the business logic consolidated, we've refactored Studio to instantiate the CLI when creating, updating, and deleting preview sites.
+
+## Data flow
+
+1. When calling the CLI:
+    - `yargs` is used to parse commands and options and to auto-generate help pages.
+    - The appropriate command is called.
+    - Progress is pretty-printed and the command runs until completion or failure.
+
+2. When Studio instantiates the CLI:
```

Comment:
Nit: Should this part be consolidated with the `Studio calling the CLI` section. Maybe it's just me but when reading "data flow" I expected it to be focused on the CLI.

Reply:
To my mind, this part describes the data flow at a high level, whereas the `Studio calling the CLI` section gets into more of the why and the upsides and downsides.